src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +18,422 @@ internal class CodeStylePage : AbstractOptionPage
+                }
+                else if (curSetting == LabelPositionOptions.OneLess)
+                {
+                    editorconfig.AppendLine("one_less_than_current");
+                }
+                else if (curSetting == LabelPositionOptions.NoIndent)
+                {
+                    editorconfig.AppendLine("no_change");
+                }
+                else
+                {
+                    throw new NotSupportedException();
+                }
+            }
+        }
+
+        private static void CSharpFormattingOptions_GenerateEditorconfig(OptionSet optionSet, Option<BinaryOperatorSpacingOptions> option, StringBuilder editorconfig)
+        {
+            var element = option.StorageLocations.OfType<EditorConfigStorageLocation<BinaryOperatorSpacingOptions>>().FirstOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+                var curSetting = optionSet.GetOption(option);
+                if (curSetting == BinaryOperatorSpacingOptions.Single)
+                {
+                    editorconfig.AppendLine("before_and_after");
+                }
+                else if (curSetting == BinaryOperatorSpacingOptions.Remove)
+                {
+                    editorconfig.AppendLine("none");
+                }
+                else if (curSetting == BinaryOperatorSpacingOptions.Ignore)
+                {
+                    editorconfig.AppendLine("ignore");
+                }
+                else
+                {
+                    throw new NotSupportedException();
+                }
+            }
+        }
+
+        private static void CSharpFormattingOptions_GenerateEditorconfig(OptionSet optionSet, StringBuilder editorconfig)
+        {
+            var element = CSharpFormattingOptions.SpaceWithinOtherParentheses.StorageLocations.OfType<EditorConfigStorageLocation<bool>>().FirstOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var firstRule = true;
+                if (optionSet.GetOption(CSharpFormattingOptions.SpaceWithinOtherParentheses))
+                {
+                    editorconfig.Append("control_flow_statements");
+                    firstRule = false;
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.SpaceWithinExpressionParentheses))
+                {
+                    if (firstRule)
+                    {
+                        firstRule = false;
+                    }
+                    else
+                    {
+                        editorconfig.Append(",");
+                    }
+
+                    editorconfig.Append("expressions");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.SpaceWithinCastParentheses))
+                {
+                    if (firstRule)
+                    {
```

Comment:
nit: this is not very pretty.. consider just building this up in a list and then using string.Join, or adding some helper function